components/gitpod-db/src/typeorm/entity/db-project.ts

```diff
@@ -26,6 +26,9 @@ export class DBProject {
   @Column()
   appInstallationId: string;
 
+  @Column()
+  config?: string;
```

Comment:
Agreed. This initial cut-through doesn't handle Dockerfile, but I was planning to look into potentially storing a custom Dockerfile as well in the near future.

(For now, I think it adds too much complexity to a simple configurator -- ideally I'd even want to remove the YAML editor entirely, and just have people click together a simple config without "scaring them away".)

Reply:
I generally agree with keeping it simple, but in this case we should think a bit ahead as we are talking about DB migrations.

Reply:
> ideally I'd even want to remove the YAML editor entirely, and just have people click together a simple config without "scaring them away

The config form should really just be a view on top of the yaml.

Reply:
What about a `JSON` with key being the path, value the content? Basically mimicking a file system overlay.

Reply:
I think this should be 
```
@Column("simple-json") 
config?: {[filePath: string]: string}
```
And on the db maybe even `mediumtext`

Reply:
Alternatively, since we only need a .gitpod.yml and a Dockerfile, we could also have something like `config?: { configString: string, dockerfile?: ImageConfigFile }`

See the `ImageConfigFile` type here:

https://github.com/gitpod-io/gitpod/blob/313fc79b91ac9fe9d17054beac363c8db43555d4/components/gitpod-protocol/src/protocol.ts#L746-L751

This would avoid all the ambiguities around `gitpod.yml` vs `.gitpod.yml` vs `gitpod.yaml` vs `.gitpod/gitpod.yaml` etc, and I think there is no need to implement a "generic file system as JSON object" here.

Reply:
We should solve that on the UI side and be flexible on the DB level. Going forward we might allow alternatives or just change the pattern.

Reply:
It also aligns with the AdditionalContent shape

Reply:
I've implemented this change, but I find the data model less elegant.

The config string is the primary thing we care about in many places, but now to get it, we have to use the "magical" `'.gitpod.yml'` key everywhere in combination with the `project.config` object, with possible typos and increased opportunities to get `undefined`.

Reply:
Fair point. We can make the type not a string but a literal type for now and either add additional values when needed or soften it to string later.

Reply:
something like this:
```
type ConfigPath = '.gitpod.yml' | '.gitpod.Dockerfile';
@Column("simple-json", { nullable: true }) 
config?: {[filePath: ConfigPath]: string}
```

Reply:
Hmm, TypeScript didn't seem to like the union type for `filePath`.

I think I'll go with this instead for now (easy to extend later):

```
  @Column("simple-json", { nullable: true })
  config?: { '.gitpod.yml': string };
```